src/main/java/de/fraunhofer/aisec/analysis/utils/Utils.java

```diff
@@ -267,6 +267,23 @@ public static String toNonQualifiedName(@NonNull String typeStr) {
 		return typeStr;
 	}
 
+	/**
+	 * Counterpart to toNonQualifiedName().
+	 *
+	 * @param fqn
+	 * @return
+	 */
+	@NonNull
+	public static String getScope(@NonNull String fqn) {
```

Comment:
Can you explain, what this method intends to achieve? Are you trying to get the namespace? What if namespaces are nested?

Reply:
Yes, this method should get "everything before the actual name of the type", i.e. a namespace in C++ and a package in Java.

`java.lang.String` -> `java.lang`
`Botan::Cipher_Mode::create` -> `Botan::Cipher_Mode`

The only purpose of this method is currently to support in the comparison of `type`s as provided by the CPG with fully qualified names of method calls in MARK. For instance, this is how it looks for Java:

```
op bla {
  java.lang.String.toString()
}
```

If the code contains a line `x.toString()`, Codyze gets the `type` of base `x` and compares it against the type given in  MARK. 
This is how it looks for C++:

```
op bla {
  Botan::Cipher_Mode::create(_,_,_,_);
}
```